### PR TITLE
fixing new assertion

### LINKED ISSUE #1: String assertion passing with partial string
This assertion:
```ts
expect(stub).to.have.callsLike(['test']);
```

and this
```ts
expect(stub).to.have.callsLike(['my test']);
```

Both passes when I call the function like this:
```ts
stub("my test that shouldn't assert");
```

### PR DESCRIPTION
**Pull request by Dodt**
Closes #1 
I verify that has a problem with the match of assert sinon, so to resolve this problem I create a new validation to make sure when it is a string will validate correctly, using expect from chai

### CODE REVIEW

File: src/lib/match.ts [hunk 1]
```diff
@@ -1,9 +1,13 @@
 import { assert } from 'sinon';
 import { getBaseTypeDiff } from './get-base-type-diff';
+import { expect } from 'chai';
 
 export function match(actual: unknown, expected: unknown) {
   let matchResult: boolean;
   try {
+    if (typeof actual === 'string' && typeof expected === 'string') {
```
Comment (Farenheith) [inline src/lib/match.ts:8]: That works, good find!
I think that maybe other types may have the same treatment and the expect function may not be needed to use. Could you do something like this:
```ts
    if (actual !== expected) {
      if (shouldFailOnExactCompare(typeof actual, typeof expected)) {
        matchResult = false;
      } else {
        assert.match(actual, expected);
      }
    }	
```

And, in the implementation of **shouldFailOnExactCompare** you could just check if both types are string

Comment (Dodt) [reply src/lib/match.ts]: you are right!! I will make this change

Comment (Dodt) [reply src/lib/match.ts]: Done!

**Comment by Farenheith** [comment]
@Dodt thanks for the contribution! All good to go!

**Pull request merged**